app/src/main/java/com/kickstarter/viewmodels/MessagesViewModel.java

```diff
@@ -250,18 +251,20 @@ public ViewModel(final @NonNull Environment environment) {
         .subscribe(this.successfullyMarkedAsRead::onNext);
 
       final Observable<List<Message>> initialMessages = initialMessageThreadEnvelope
-        .map(MessageThreadEnvelope::messages)
-        .filter(ObjectUtils::isNotNull);
+        .map(MessageThreadEnvelope::messages);
 
       final Observable<List<Message>> newMessages = sentMessageThreadEnvelope
         .map(MessageThreadEnvelope::messages);
 
+      // Concat distinct messages to initial message list. Return just the new messages if
+      // initial list is null, i.e. a new message thread.
       final Observable<List<Message>> updatedMessages = initialMessages
         .compose(takePairWhen(newMessages))
-        .map(mm -> ListUtils.concatDistinct(mm.first, mm.second));
+        .map(mm -> mm.first == null ? mm.second : ListUtils.concatDistinct(mm.first, mm.second));
```

Comment:
is the null check here redundant now that you're filtering for `isNotNull` on `initialMessages` below?